test/c/no-autoconf.yaml

```diff
@@ -0,0 +1,12 @@
+test:
+- extension
+directives:
+- domain: c
+  directive: autodoc
+  arguments:
+  - bool.c
+conf-overrides:
+  hawkmoth_clang: -nostdinc
+  hawkmoth_autoconf:
```

Comment:
I think it's this one failing in CI, though it passes locally. Don't have the time now (short lunch break). The idea was to add `-nostdinc` to normalize behaviour across distros and disable autoconf such that `stdbool.h` wouldn't be found that way either.

Not sure this one tests what we want to though, so I may just remove it: is it failing to find the header because we disabled auto configuring or because we set `-nostdinc`?